.vsts-dotnet-ci.yml

```
@@ -34,6 +34,10 @@ jobs:
       mergeTestResults: true
     continueOnError: true
     condition: always()
+  - powershell: | 
+      taskkill /f /im msbuild.exe
```

This very much looks like a workaround. Why are there hanging processes in the first place? Can we fix that? Also, this makes it a little cleaner:
```suggestion
      taskkill /f /im msbuild.exe /im vbcscompiler.exe
```

---

I didn't know you can pass /im multiple times!

---

One thing to clarify here: My expectation is that there were always processes hanging around.  I don't believe that this change modifies process lifetime.  You could try running `dotnet build-server shutdown` to kill the compiler server.  One thing to confirm - are we getting full perf logs from these runs, or are they truncated because of the call to `taskkill`?

---

If we kill the processes, the file that they generate isn't completed:

```
[2021-04-05T19:37:33.6451866Z] Event=Microsoft-Build/MSBuildExeStart ProcessID=7688 ThreadID=1	 commandLine="D:\a\1\s\artifacts\bin\Microsoft.Build.Engine.UnitTests\Debug\net472\MSBuild.exe /nologo /nodemode:1 /nodeReuse:true /low:false" 
[2021-04-05T19:37:34.0831946Z] Event=Microsoft-Build/RequestThreadProcStart ProcessID=7688 ThreadID=11	 
[2021-04-05T19:37:34.0872085Z] Event=Microsoft-Build/BuildProjectStart ProcessID=7688 ThreadID=11	 projectPath="C:\Users\VssAdministrator\AppData\Local\Temp\041qbmta.yej\tmp4965ff255b434b4e8c54fa1b6dc9cf10.tmp" 
[2021-04-05T19:37:34.0962009Z] Event=Microsoft-Build/LoadDocumentStart ProcessID=7688 ThreadID=11	 fullPath="C:\Users\VssAdministrator\AppData\Local\Temp\041qbmta.yej\tmp4965ff255b434b4e8c54fa1b6dc9cf10.tmp" 
[2021-04-05T19:37:34.1042039Z] Event=Microsoft-Build/LoadDocumentStop ProcessID=7688 ThreadID=11	 fullPath="C:\Users\VssAdministrator\AppData\Local\Temp\041qbmta.yej\tmp4965ff255b434b4e8c54fa1b6dc9cf10.tmp" 
[2021-04-05T19:37:34.1072967Z] Event=Microsoft-Build/ParseStart ProcessID=7688 ThreadID=11	 projectFileName="C:\Users\VssAdministrator\AppData\Local\Temp\041qbmta.yej\tmp4965ff255b434b4e8c54fa1b6dc9cf10.tmp" 
[2021-04-05T19:37:34.1142095Z] Event=Microsoft-Build/ParseStop ProcessID=7688 ThreadID=11	 projectFileName="C:\Users\VssAdministrator\AppData\Local\Temp\041qbmta.yej\tmp4965ff255b434b4e8c54fa1b6dc9cf10.tmp" 
[2021-04-05T19:37:34.3042076Z] Event=Microsoft-Build/TargetStart ProcessID=7688 ThreadID=11	 targetName="msbuild" 
[2021-04-05T19:37:34.3582095Z] Event=Microsoft-Build/LoadDocumentStart ProcessID=7688 ThreadID=11	 fullPath="D:\a\1\s\artifacts\bin\Microsoft.Build.Engine.UnitTests\Debug\net472\Microsoft.Common.tasks" 
[2021-04-05T19:37:34.3602064Z] Event=Microsoft-Build/LoadDocumentStop ProcessID=7688 ThreadID=11	 fullPath="D:\a\1\s\artifacts\bin\Microsoft.Build.Engine.UnitTests\Debug\net472\Microsoft.Common.tasks" 
[2021-04-05T19:37:34.3602064Z] Event=Microsoft-Build/ParseStart ProcessID=7688 ThreadID=11	 projectFileName="D:\a\1\s\artifacts\bin\Microsoft.Build.Engine.UnitTests\Debug\net472\Microsoft.Common.tasks" 
[2021-04-05T19:37:34.3621959Z] Event=Microsoft-Build/ParseStop ProcessID=7688 ThreadID=11	 projectFileName="D:\a\1\s\artifacts\bin\Microsoft.Build.Engine.UnitTests\Debug\net472\Microsoft.Common.tasks" 
[2021-04-05T19:37:34.4402082Z] Event=Microsoft-Build/TargetStop ProcessID=7688 ThreadID=11	 targetName="msbuild" 
[2021-04-05T19:37:34.4442095Z] Event=Microsoft-Build/BuildProjectStop ProcessID=7688 ThreadID=11	 projectPath="C:\Users\VssAdministrator\AppData\Local\Temp\041qbmta.yej\tmp4965ff255b434b4e8c54fa1b6dc9cf10.tmp" targets="msbuild" 
[2021-04-05T19:37:34.4462038Z] Event=Microsoft-Build/RequestThreadProcStop ProcessID=7688 ThreadID=11	 
[2021-04-05T19:37:34.5532097Z] Event=Microsoft-Build/RequestThreadProcStart ProcessID=7688 ThreadID=11	 
[2021-04-05T19:37:34.5532097Z] Event=Microsoft-Build/BuildProjectStart ProcessID=7688 ThreadID=11	 projectPath="C:\Users\VssAdministrator\AppData\Local\Temp\041qbmta.yej\tmp4965ff255b434b4e8c54fa1b6dc9cf10.tmp" 
[2021-04-05T19:37:34.5571985Z] Event=Microsoft-Build/TargetStart ProcessID=7688 ThreadID=11	 targetName="msbuild" 
[2021-04-05T19:37:34.5612210Z] Event=Microsoft-Build/TargetStop ProcessID=7688 ThreadID=11	 targetName="msbuild" 
[2021-04-05T19:37:34.5612210Z] Event=Microsoft-Build/BuildProjectStop ProcessID=7688 ThreadID=11	 projectPath="C:\Users\VssAdministrator\AppData\Local\Temp\041qbmta.yej\tmp4965ff255b434b4e8c54fa1b6dc9cf10.tmp" targets="msbuild" 
[2021-04-05T19:37:34.5612210Z] Event=Microsoft-Build/RequestThreadProcStop ProcessID=7688 ThreadID=11	 
[2021-04-05T19:40:57.3420042Z] Event=Microsoft-Build/RequestThreadProcStart ProcessID=7688 ThreadID=11	 
[2021-04-05T19:40:57.3420042Z] Event=Microsoft-Build/BuildProjectStart ProcessID=7688 ThreadID=11	 projectPath="C:\Users\VssAdministrator\AppData\Local\Temp\041qbmta.yej\tmpe37eba57a280484fb0a6015308bebaff.tmp" 
[2021-04-05T19:40:57.3430119Z] Event=Microsoft-Build/LoadDocumentStart ProcessID=7688 ThreadID=11	 fullPath="C:\Users\VssAdministrator\App
```
Notice how it ends on `LoadDocumentStart` and has no `MSBuildExeStop`

I'll try `dotnet build-server` shutdown, which I didn't realize was an option!

---

If certain events hadn't fired, wouldn't that still mean those events wouldn't fire as it tries to get out of the build as fast as possible?

---

Right. I'd like to get this in a state that is "good enough" if we want to see this perf data and act on it. Even if it's not the perfect solution. Part of me suspects digging into this node issue could take longer than expected. I've been digging into it without much result so far. The current idea is to set the MSBUILDDISABLENODEREUSE env var to 1 and see if the node will realize it needs to die.